actioncable/app/javascript/action_cable/subscriptions.js

```diff
@@ -29,10 +29,15 @@ export default class Subscriptions {
   // Private
 
   add(subscription) {
+    const existing = this.findAll(subscription.identifier).length > 0
     this.subscriptions.push(subscription)
     this.consumer.ensureActiveConnection()
     this.notify(subscription, "initialized")
-    this.subscribe(subscription)
+    if (!existing) {
+      this.subscribe(subscription)
+    } else if (!this.guarantor.pendingSubscriptions.some((s) => s.identifier === subscription.identifier)) {
+      this.notify(subscription, "connected")
```

Comment:
Why `connected`? If it's pending, it could also be rejected as well (or left without any response from the server); we shouldn't mark it as connected here.